lib/ansible/plugins/connection/paramiko_ssh.py

```diff
@@ -399,53 +409,108 @@ def exec_command(self, cmd, in_data=None, sudoable=True):
 
         cmd = to_bytes(cmd, errors='surrogate_or_strict')
 
-        no_prompt_out = b''
-        no_prompt_err = b''
-        become_output = b''
+        event = threading.Event()
+        chan.in_buffer.set_event(event)
+        chan.in_stderr_buffer.set_event(event)
 
-        try:
-            chan.exec_command(cmd)
-            if self.become and self.become.expect_prompt():
-                passprompt = False
-                become_sucess = False
-                while not (become_sucess or passprompt):
-                    display.debug('Waiting for Privilege Escalation input')
+        b_stdout = b_stderr = b''
+        b_tmp_stdout = b_tmp_stderr = b''
 
+        chan.exec_command(cmd)
+        if self.become and self.become.expect_prompt():
+            while not (self._flags['become_success'] or chan.exit_status_ready()):
+                event.wait()
+
+                if chan.recv_ready():
                     chunk = chan.recv(bufsize)
-                    display.debug("chunk is: %s" % chunk)
-                    if not chunk:
-                        if b'unknown user' in become_output:
-                            raise AnsibleError('user %s does not exist' % self._play_context.become_user)
-                        else:
-                            break
-                            # raise AnsibleError('ssh connection closed waiting for password prompt')
-                    become_output += chunk
-
-                    # need to check every line because we might get lectured
-                    # and we might get the middle of a line in a chunk
-                    for l in become_output.splitlines(True):
-                        if self.become.check_success(l):
-                            become_sucess = True
-                            break
-                        elif self.become.check_password_prompt(l):
-                            passprompt = True
-                            break
-
-                if passprompt:
+                    display.debug("stdout chunk is: %s" % chunk)
+                    b_tmp_stdout += chunk
+
+                    b_output, b_unprocessed = self._examine_output('stdout', b_tmp_stdout, sudoable)
```

Comment:
Do you really need this extra var?
Maybe just save stuff there right away?
```suggestion
                    b_output, b_tmp_stdout = self._examine_output('stdout', b_tmp_stdout, sudoable)
```